Snackbar/Snackbar.jsx

```diff
@@ -23,6 +31,11 @@ export default class Snackbar extends MaterialComponent {
   componentWillUnmount() {
     this.MDComponent.destroy && this.MDComponent.destroy();
   }
+  shouldComponentUpdate(props, state) {
+    return (
+      shallowDiffers(this.props, props) || shallowDiffers(this.state, state)
```

Comment:
doesn't preact do this on its own?

Reply:
Not in `preact`, but in `preact-compat`.
That's why I said importing and inheriting from `PureComponent` from `preact-compat` may also solve this problem. However I'm not sure if it worth it importing hundreds of lines of codes when using only less than 10 lines of them.
Probably webpack tree shaking can help though.

reference:
https://github.com/developit/preact/blob/0a88752e4890d391b5e23809d9432f625a833ae8/src/vdom/component.js#L83

https://github.com/developit/preact-compat/blob/5ca57489b75c539a0ad022f3624d97d755c939b5/src/index.js#L591